src/ps/descriptor.js

```
@@ -504,6 +539,102 @@ define(function (require, exports, module) {
     };
 
     /**
+     * Efficiently get a set of properties on an arbitrary set of references.
+     * 
+     * @param {Array.<object|string|Array.<object>>} references
+     * @param {Array.<string>} properties
+     * @param {object=} options Use continueOnError to allow some properties to
+     *  not be returned
+     * @return {Promise.<Array.<Object.<string, *>>>}
+     */
+    Descriptor.prototype.batchMultiGetProperties = function (references, properties, options) {
+        if (properties.length === 0) {
+            return Promise.resolve([{}]);
+        }
+
+        if (options === undefined) {
+            options = {};
+        }
+
+        var multiGetOptions = {
+            useMultiGet: true,
+            failOnMissingProperty: !options.continueOnError
+        };
+
+        var commands = references.map(function (reference) {
+            var descriptor = {
+                null: _wrap(reference, properties)
+            };
+
+            return {
+                name: "get",
+                descriptor: descriptor,
+                options: multiGetOptions
+            };
+        });
+
+        return this.batchPlay(commands, options)
+            .then(function (response) {
+                if (options.continueOnError) {
```

No, I think it's correct. If `continueOnError` is set, `this.batchPlay` returns a [results, errors] array, and we're intentionally ignoring the errors here. If it's not set, `this.batchPlay` just returns the results directly (assuming there was no error).